sporkfish/searcher/pvs.py

```diff
@@ -0,0 +1,250 @@
+from typing import Optional, Tuple
+
+import chess
+
+from sporkfish.board.board import Board
+from sporkfish.evaluator.evaluator import Evaluator
+from sporkfish.searcher.minimax import MiniMaxVariants
+from sporkfish.searcher.move_ordering.move_orderer import MoveOrderer
+from sporkfish.searcher.searcher_config import SearcherConfig
+from sporkfish.statistics import NodeTypes
+from sporkfish.zobrist_hasher import ZobristStateInfo
+
+
+class PVSSp(MiniMaxVariants):
+    def __init__(
+        self,
+        evaluator: Evaluator,
+        searcher_config: SearcherConfig = SearcherConfig(),
+    ) -> None:
+        super().__init__(evaluator, searcher_config)
+
+    def _pvs(
+        self,
+        board: Board,
+        depth: int,
+        alpha: float,
+        beta: float,
+        zobrist_state: Optional[ZobristStateInfo],
+    ) -> float:
+        """
+        Principal Variation Search implementation with alpha-beta pruning. For non-root nodes.
+
+        Args:
+            board: The current state of the chess board.
+            depth: The remaining depth to search.
+            alpha: The alpha value for alpha-beta pruning.
+            beta: The beta value for alpha-beta pruning.
+
+        Returns:
+            The evaluation score of the current board position.
+        """
+        value = -float("inf")
+
+        # Base case: devolve to quiescence search
+        # We currently only expect max 4 captures to reach a quiet (non-capturing) position
+        # This is not ideal, but otherwise the search becomes incredibly slow
+        if depth == 0:
+            return self._quiescence(board, 4, alpha, beta, zobrist_state)
+
+        # Probe the transposition table for an existing entry
+        if zobrist_state and (
+            tt_entry := self._transposition_table.probe(
+                zobrist_state.zobrist_hash, depth
+            )
+        ):
+            self._statistics.increment_nodes_from_tt()
+            return tt_entry["score"]  # type: ignore
+
+        self._statistics.increment_node_visited(NodeTypes.NEGAMAX)
```

Comment:
I think the naming is fine for now bercause PVS is a negamax family algorithm but if needed we can rename these NodeTypes.Search - up to @ccjeremylo

Reply:
I must have missed it sorry - my money is on that the differentiation doesn't need to exist since we won't have multiple searchers running at the same time, that would be very slow. Anyhow it doesn't affect this here yet, so for future.

Reply:
Good point